library/public/image.h

```diff
@@ -49,27 +86,41 @@ class F3D_EXPORT image
   ///@{ @name Resolution
   /**
    * Set/Get image resolution.
+   *
+   * \deprecated { setResolution is deprecated, use the appropriate constructor }
    */
   unsigned int getWidth() const;
   unsigned int getHeight() const;
-  image& setResolution(unsigned int width, unsigned int height);
+#ifndef F3D_NO_DEPRECATED
+  F3D_DEPRECATED image& setResolution(unsigned int width, unsigned int height);
+#endif
   ///@}
 
   ///@{ @name Channel Count
   /**
    * Set/Get image channel count.
+   *
+   * \deprecated { setChannelCount is deprecated, use the appropriate constructor }
    */
   unsigned int getChannelCount() const;
-  image& setChannelCount(unsigned int dim);
+#ifndef F3D_NO_DEPRECATED
+  F3D_DEPRECATED image& setChannelCount(unsigned int dim);
+#endif
   ///@}
 
+  /**
+   * Get image channel type.
+   * throw an image::read_exception if the type is unknown.
+   */
+  ChannelType getChannelType() const;
+
   ///@{ @name Buffer Data
   /**
    * Set/Get image buffer data.
-   * Its size is expected to be `width * height * channelCount`.
+   * Its size is expected to be `width * height * channelCount * typeSize`.
    */
-  image& setData(unsigned char* buffer);
-  unsigned char* getData() const;
+  image& setData(void* buffer);
+  void* getData() const;
```

Comment:
is this breaking retrocompat ? I dont think so apart from a potential cast warning ?

Reply:
Ah yes indeed. People using `getData` will have to cast the returned pointer to the expected pointer type

Reply:
I'll keep `unsigned char*` functions but flag them as deprecated